components/apimgt/org.wso2.carbon.apimgt.rest.api.dcr/src/main/java/org/wso2/carbon/apimgt/rest/api/dcr/web/impl/RegistrationServiceImpl.java

```diff
@@ -353,26 +362,38 @@ private OAuthApplicationInfo createApplication(String applicationName, OAuthAppR
             serviceProvider.setApplicationName(applicationName);
             serviceProvider.setDescription("Service Provider for application " + appName);
             serviceProvider.setSaasApp(applicationInfo.getIsSaasApplication());
-            ServiceProviderProperty[] serviceProviderProperties = new ServiceProviderProperty[4];
+
+            // convert to list
+            List<ServiceProviderProperty> serviceProviderProperties = new ArrayList<>();
+
             ServiceProviderProperty serviceProviderProperty = new ServiceProviderProperty();
             serviceProviderProperty.setName(APP_DISPLAY_NAME);
             serviceProviderProperty.setValue(applicationName);
-            serviceProviderProperties[0] = serviceProviderProperty;
+            serviceProviderProperties.add(serviceProviderProperty);
             ServiceProviderProperty tokenTypeProviderProperty = new ServiceProviderProperty();
             tokenTypeProviderProperty.setName(APIConstants.APP_TOKEN_TYPE);
             tokenTypeProviderProperty.setValue(applicationInfo.getTokenType());
-            serviceProviderProperties[1] = tokenTypeProviderProperty;
+            serviceProviderProperties.add(tokenTypeProviderProperty);
             ServiceProviderProperty consentProperty = new ServiceProviderProperty();
             consentProperty.setDisplayName(APIConstants.APP_SKIP_CONSENT_DISPLAY);
             consentProperty.setName(APIConstants.APP_SKIP_CONSENT_NAME);
             consentProperty.setValue(APIConstants.APP_SKIP_CONSENT_VALUE);
-            serviceProviderProperties[2] = consentProperty;
+            serviceProviderProperties.add(consentProperty);
             ServiceProviderProperty logoutConsentProperty = new ServiceProviderProperty();
             logoutConsentProperty.setDisplayName(APIConstants.APP_SKIP_LOGOUT_CONSENT_DISPLAY);
             logoutConsentProperty.setName(APIConstants.APP_SKIP_LOGOUT_CONSENT_NAME);
             logoutConsentProperty.setValue(APIConstants.APP_SKIP_LOGOUT_CONSENT_VALUE);
-            serviceProviderProperties[3] = logoutConsentProperty;
-            serviceProvider.setSpProperties(serviceProviderProperties);
+            serviceProviderProperties.add(logoutConsentProperty);
+
+            if (!orgId.isEmpty()) {
```

Comment:
might break if orgId is not sent. (or using basic auth)
```suggestion
            if (StringUtils.isNotBlank(orgId)) {
```